poly/src/polynomial/multivariate/sparse.rs

```diff
@@ -136,19 +136,22 @@ impl<F: Field> DenseMVPolynomial<F> for SparsePolynomial<F, SparseTerm> {
         terms.sort_by(|(_, t1), (_, t2)| t1.cmp(t2));
         // If any terms are duplicated, add them together
         let mut terms_dedup: Vec<(F, SparseTerm)> = Vec::new();
-        for term in terms {
-            if let Some(prev) = terms_dedup.last_mut() {
-                if prev.1 == term.1 {
-                    *prev = (prev.0 + term.0, prev.1.clone());
-                    continue;
-                }
-            };
+        for (coeff, term) in terms {
             // Assert correct number of indeterminates
             assert!(
-                term.1.iter().all(|(var, _)| *var < num_vars),
+                term.iter().all(|(var, _)| *var < num_vars),
                 "Invalid number of indeterminates"
             );
```

Comment:
This assertion is moved to the beginning of the loop because we must check this at the start of the loop before doing any calculation (this allows us to panic earlier if necessary)